wren-ui/src/components/Deploy.tsx

```diff
@@ -0,0 +1,65 @@
+import { Button, Space, Typography } from 'antd';
+import CheckCircleOutlined from '@ant-design/icons/CheckCircleOutlined';
+import LoadingOutlined from '@ant-design/icons/LoadingOutlined';
+import WarningOutlined from '@ant-design/icons/WarningOutlined';
+import {
+  useDeployMutation,
+  useDeployStatusQuery,
+} from '@/apollo/client/graphql/deploy.generated';
+import { SyncStatus } from '@/apollo/client/graphql/__types__';
+
+const { Text } = Typography;
+
+const getDeployStatus = (deploying: boolean, status: SyncStatus) => {
+  const syncStatus = deploying ? SyncStatus.IN_PROGRESS : status;
+
+  return (
+    {
+      [SyncStatus.IN_PROGRESS]: (
+        <Space size={[4, 0]}>
+          <LoadingOutlined className="mr-1 gray-1" />
+          <Text className="gray-1">Deploying...</Text>
+        </Space>
+      ),
+      [SyncStatus.SYNCRONIZED]: (
+        <Space size={[4, 0]}>
+          <CheckCircleOutlined className="mr-1 green-7" />
+          <Text className="gray-1">Synced</Text>
+        </Space>
+      ),
+      [SyncStatus.UNSYNCRONIZED]: (
+        <Space size={[4, 0]}>
+          <WarningOutlined className="mr-1 gold-6" />
+          <Text className="gray-1">Undeployed changes</Text>
+        </Space>
+      ),
+    }[syncStatus] || ''
+  );
+};
+
+export default function Deploy() {
+  const [deployMutation, { loading: deploying }] = useDeployMutation();
+  const { data, loading } = useDeployStatusQuery({
+    pollInterval: 1000,
+  });
```

Comment:
I have some concerns about long polling with one sec.
It seems like we will infinite polling from the page start to end, does it really need?

If it is necessary, I will suggest we can add a Visibility API to have a switch of it.

Reply:
According to the results of the previous discussion, since there are not many trigger behaviors for changing MDL, I first adjusted this version - Use trigger deployStatusQueryResult.refetch() for actions instead of the polling method to get the latest sync status.

I think... we can go through how to implement better ways if the product has complex behavior in the future.